c7n/handler.py

```diff
@@ -139,7 +139,8 @@ def init_config(policy_config):
 
 def dispatch_event(event, context):
     # default event.detail for EB Scheduler is '{}', not {}
-    event['detail'] = {} if event.get('detail') == '{}' else event['detail']
+    if 'detail' in event and event.get('detail') == '{}':
```

Comment:
not quite.. simplifying/correcting the `else` expression is event.get('detail', {})

Reply:
Switched back and followed your suggestion with the else clause.